QC.wdl

```diff
@@ -1,88 +1,81 @@
 version 1.0
 # Copyright 2018 Sequencing Analysis Support Core - Leiden University Medical Center
 
-import "QualityReport.wdl" as QR
 import "tasks/cutadapt.wdl" as cutadapt
 import "tasks/common.wdl" as common
+import "tasks/fastqc.wdl" as fastqc
 
 workflow QC {
     input {
         File read1
         File? read2
         String outputDir
-        Boolean alwaysRunAdapterClipping = false
+        # Adapters and contaminations are optional and need at least one item if defined.
+        # This is necessary so no empty flags are used in cutadapt.
+        Array[String]+? adapters = ["AGATCGGAAGAG"]  # Illumina universal adapter
+        Array[String]+? contaminations
         Int minimumReadLength = 2 # Choose 2 here to compensate for cutadapt weirdness. I.e. Having empty or non-sensical 1 base reads.
-
+        # A readgroupName so cutadapt creates a unique report name. This is useful if all the QC files are dumped in one folder.
+        String readgroupName = sub(basename(read1),"(\.fq)?(\.fastq)?(\.gz)?", "")
         Map[String, String] dockerTags = {"fastqc": "0.11.7--4",
             "biopet-extractadaptersfastqc": "0.2--1", "cutadapt": "1.16--py36_2"}
+
     }
 
-    String read1outputDir = outputDir + "/QC/read1"
-    String read2outputDir = outputDir + "/QC/read2"
-    String read1outputDirAfterQC = outputDir + "/QCafter/read1"
-    String read2outputDirAfterQC = outputDir + "/QCafter/read2"
-    String adapterClippingOutputDir = outputDir + "/AdapterClipping"
-    String seqstatBeforeFile = outputDir + "/QC/seqstat.json"
-    String seqstatAfterFile = outputDir + "/QCafter/seqstat.json"
+    Boolean runAdapterClipping = length(select_first([adapters, []])) + length(select_first([contaminations, []])) > 0
 
-    call QR.QualityReport as qualityReportRead1 {
+    call fastqc.Fastqc as FastqcRead1 {
         input:
-            read = read1,
-            outputDir = read1outputDir,
-            dockerTags = dockerTags
+            seqFile = read1,
+            outdirPath = outputDir + "/",
+            dockerTag = dockerTags["fastqc"]
     }
 
     if (defined(read2)) {
-        call QR.QualityReport as qualityReportRead2 {
+        call fastqc.Fastqc as FastqcRead2 {
             input:
-                read = select_first([read2]),
-                outputDir = read2outputDir,
-                dockerTags = dockerTags
+                seqFile = select_first([read2]),
+                outdirPath = outputDir + "/",
+                dockerTag = dockerTags["fastqc"]
         }
+        String read2outputPath = outputDir + "/cutadapt_" + basename(select_first([read2]))
     }
 
-    # if no adapters are found, why run cutadapt? Unless cutadapt is used for quality trimming.
-    # In which case alwaysRunCutadapt can be set to true by the user.
-    Boolean runAdapterClipping = defined(qualityReportRead1.adapters) || defined(qualityReportRead2.adapters) || alwaysRunAdapterClipping
-
     if (runAdapterClipping) {
```

Comment:
Shouldn't this be `defined(adapters)`?

Reply:
It should also be running if contaminations are defined.

Reply:
I seem to have overlooked the definition of `runAdapterClipping`